registries/adapters/local_openshift_adapter.go

```diff
@@ -69,87 +71,55 @@ func (r LocalOpenShiftAdapter) GetImageNames() ([]string, error) {
+	for _, ns := range r.Config.Namespaces {
+		is, err := imageClient.ImageStreams(ns).List(meta_v1.ListOptions{})
+		if err != nil {
+			log.Errorf("Failed to get list of imagestreams for namespace [%v]: %v", ns, err)
+			continue
+		}
+		for _, i := range is.Items {
+			imageList = append(imageList, fmt.Sprintf("%v/%v", ns, i.Name))
+		}
 	}
-
 	return imageList, nil
 }
 
 // FetchSpecs - retrieve the spec for the image names.
 func (r LocalOpenShiftAdapter) FetchSpecs(imageNames []string) ([]*bundle.Spec, error) {
 	log.Debug("LocalOpenShiftAdapter::FetchSpecs")
 	specList := []*bundle.Spec{}
-	registryIP, err := r.getServiceIP("docker-registry", "default")
-	if err != nil {
-		log.Errorf("Failed get docker-registry service information.")
-		return nil, err
-	}
 
 	openshiftClient, err := clients.Openshift()
 	if err != nil {
 		log.Errorf("Failed to instantiate OpenShift client.")
 		return nil, err
 	}
-
-	listImages, err := openshiftClient.ListRegistryImages()
-	if err != nil {
-		log.Errorf("Failed to load registry images")
-		return nil, err
-	}
-
-	for _, image := range listImages.Items {
-		n := strings.Split(image.DockerImageManifest, "@")[0]
-		for _, providedImage := range imageNames {
-			if providedImage == n {
-				spec, err := r.loadSpec(image)
-				if err != nil {
-					log.Errorf("Failed to load image spec")
-					continue
-				}
-				if strings.HasPrefix(n, registryIP) == false {
-					log.Debugf("Image does not have a registry IP as prefix. This might cause problems but not erroring out.")
-				}
-				if r.Config.Namespaces == nil {
-					log.Debugf("Namespace not set. Assuming `openshift`")
-					r.Config.Namespaces = append(r.Config.Namespaces, "openshift")
-				}
-				spec.Image = n
-				nsList := strings.Split(n, "/")
-				var namespace string
-				if len(nsList) == 0 {
-					log.Errorf("Image [%v] is not in the proper format. Erroring.", n)
-					continue
-				} else if len(nsList) < 3 {
-					// Image does not have any registry prefix. May be a product of S2I
-					// Expecting openshift/foo-bundle
-					namespace = nsList[0]
-				} else {
-					// Expecting format: 172.30.1.1:5000/openshift/foo-bundle
-					namespace = nsList[1]
-				}
-				for _, ns := range r.Config.Namespaces {
-					// logging to warn users about the potential bug if
-					// the svc-acct does not have access to the namespace.
-					if ns != "openshift" {
-						log.Warningf("You may not be able to load provision images from the namespace: %v.\n"+
-							"You should make sure that the namespace has given the permissions for the "+
-							"system:authenticated group.", ns)
-					}
-					if ns == namespace {
-						log.Debugf("Image [%v] is in configured namespace [%v]. Adding to SpecList.", n, ns)
-						specList = append(specList, spec)
-					}
-				}
-			}
+	imageClient := openshiftClient.Image()
+
+	for _, image := range imageNames {
+		fullName := strings.Split(image, "/")
+		ns := fullName[0]
+		iName := fullName[1]
```

Comment:
Array length check needed here? Or do we have high confidence that the data we're ingesting is formatted this way

Reply:
The godoc states:

> func Split(s, sep string) []string
>
> Split slices s into all substrings separated by sep and returns a slice of the substrings between those separators.
>
> If s does not contain sep and sep is not empty, Split returns a slice of length 1 whose only element is s.
>
> If sep is empty, Split splits after each UTF-8 sequence. If both s and sep are empty, Split returns an empty slice.
>
> It is equivalent to SplitN with a count of -1. 

Just tried it, if the string does not have a "/" in it, you won't get a second element in the array it will cause the broker to panice.